File: exercise/php/day03/tests/SantaWorkshopServiceTest.php
<?php

declare(strict_types = 1);

use Preparation\Gift;
use Preparation\SantaWorkshopService;

const RECOMMENDED_AGE = 'recommendedAge';

beforeEach(function (): void {
	$this->service = new SantaWorkshopService;
});

it('prepares a valid gift', function (): void {
	$gift = $this->service->prepareGift('Bitzee', 3, 'Purple', 'Plastic');

	expect($gift)->toBeInstanceOf(Gift::class);
});

it('retrieves attribute on gift', function (): void {
	$gift = $this->service->prepareGift('Furby', 1, 'Multi', 'Cotton');
	$gift->addAttribute(RECOMMENDED_AGE, '3');

	expect($gift->recommendedAge())->toBe(3);
});

it('fails for a too heavy gift', function (): void {
	$prepareGift = fn () => $this->service->prepareGift('Dog-E', 6, 'White', 'Metal');

	expect($prepareGift)->toThrow(InvalidArgumentException::class, 'Gift is too heavy for Santa\'s sleigh');
});

File: exercise/kotlin/day03/src/test/kotlin/SantaWorkshopServiceTest.kt
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe
import preparation.Gift
import preparation.SantaWorkshopService

class SantaWorkshopServiceTest : DescribeSpec({
    val service = SantaWorkshopService()

    describe("prepareGift") {
        it("should prepare a gift with valid parameters") {
            val giftName = "Bitzee"
            val weight = 3.0
            val color = "Purple"
            val material = "Plastic"

            service.prepareGift(giftName, weight, color, material) shouldNotBe null
        }

        it("should retrieve an attribute to a gift") {
            val giftName = "Furby"
            val weight = 1.0
            val color = "Multi"
            val material = "Cotton"

            val gift = Gift(giftName, weight, color, material)
            gift.addAttribute("recommendedAge", "3")

            gift.getRecommendedAge() shouldBe 3
        }

        it("should fail if gift is too heavy") {
            val giftName = "Dog-E"
            val weight = 6.0
            val color = "White"
            val material = "Metal"

            shouldThrow<IllegalArgumentException> {
                service.prepareGift(giftName, weight, color, material)
            }.message shouldBe "Gift is too heavy for Santa's sleigh"
        }
    }
})